src/lint/practice_exercises.nim

```diff
@@ -0,0 +1,64 @@
+import std/[json, os]
+import ".."/helpers
+import "."/validators
+
+proc isValidAuthorOrContributor(data: JsonNode, context: string, path: string): bool =
+  if isObject(data, context, path):
+    result = true
+    if not checkString(data, "github_username", path):
+      result = false
+    if not checkString(data, "exercism_username", path, isRequired = false):
+      result = false
+
+proc checkFiles(data: JsonNode, context, path: string): bool =
+  result = true
+  if hasObject(data, context, path):
+    if not checkArrayOfStrings(data, context, "solution", path):
+      result = false
+    if not checkArrayOfStrings(data, context, "test", path):
+      result = false
+    if not checkArrayOfStrings(data, context, "example", path):
+      result = false
+  else:
+    result = false
+
+proc isValidPracticeExerciseConfig(data: JsonNode, path: string): bool =
+  if isObject(data, "root", path):
+    result = true
+    if not checkArrayOf(data, "authors", path, isValidAuthorOrContributor):
+      result = false
+    if not checkArrayOf(data, "contributors", path, isValidAuthorOrContributor,
+                        isRequired = false):
+      result = false
+    if not checkFiles(data, "files", path):
+      result = false
+    if not checkString(data, "language_versions", path, isRequired = false):
+      result = false
+
+proc isEveryPracticeExerciseConfigValid*(trackDir: string): bool =
```

Comment:
And again, this logic is very similar to the concept exercises logic. We could extract it to a helper method.